File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/SpanDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Span details
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=SpanDetails.Builder.class)

public final class SpanDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"parentTraceId", "spanId", "parentSpanId", "startTime", "endTime", "kind", "spanName", "attributes", "events", "status"})
    public SpanDetails(String parentTraceId, String spanId, String parentSpanId, Long startTime, Long endTime, String kind, String spanName, java.util.Map<String, Object> attributes, java.util.List<TraceEvent> events, SpanStatus status) {
        super();
        this.parentTraceId = parentTraceId;
        this.spanId = spanId;
        this.parentSpanId = parentSpanId;
        this.startTime = startTime;
        this.endTime = endTime;
        this.kind = kind;
        this.spanName = spanName;
        this.attributes = attributes;
        this.events = events;
        this.status = status;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * trace id
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parentTraceId")
private String parentTraceId;

        /**
         * trace id
         * @param parentTraceId the value to set
         * @return this builder
         **/
        

public Builder parentTraceId(String parentTraceId) {
    this.parentTraceId = parentTraceId;
    return this;
}
            /**
     * span id
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("spanId")
private String spanId;

        /**
         * span id
         * @param spanId the value to set
         * @return this builder
         **/
        

public Builder spanId(String spanId) {
    this.spanId = spanId;
    return this;
}
            /**
     * parent span id
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parentSpanId")
private String parentSpanId;

        /**
         * parent span id
         * @param parentSpanId the value to set
         * @return this builder
         **/
        

public Builder parentSpanId(String parentSpanId) {
    this.parentSpanId = parentSpanId;
    return this;
}
            /**
     * start time
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("startTime")
private Long startTime;

        /**
         * start time
         * @param startTime the value to set
         * @return this builder
         **/
        

public Builder startTime(Long startTime) {
    this.startTime = startTime;
    return this;
}
            /**
     * end time
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("endTime")
private Long endTime;

        /**
         * end time
         * @param endTime the value to set
         * @return this builder
         **/
        

public Builder endTime(Long endTime) {
    this.endTime = endTime;
    return this;
}
            /**
     * type of span
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("kind")
private String kind;

        /**
         * type of span
         * @param kind the value to set
         * @return this builder
         **/
        

public Builder kind(String kind) {
    this.kind = kind;
    return this;
}
            /**
     * span name
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("spanName")
private String spanName;

        /**
         * span name
         * @param spanName the value to set
         * @return this builder
         **/
        

public Builder spanName(String spanName) {
    this.spanName = spanName;
    return this;
}
            /**
     * span attributes
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("attributes")
private java.util.Map<String, Object> attributes;

        /**
         * span attributes
         * @param attributes the value to set
         * @return this builder
         **/
        

public Builder attributes(java.util.Map<String, Object> attributes) {
    this.attributes = attributes;
    return this;
}
            /**
     * collections of trace objects
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("events")
private java.util.List<TraceEvent> events;

        /**
         * collections of trace objects
         * @param events the value to set
         * @return this builder
         **/
        

public Builder events(java.util.List<TraceEvent> events) {
    this.events = events;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("status")
private SpanStatus status;



public Builder status(SpanStatus status) {
    this.status = status;
    return this;
}


        public SpanDetails build() {
            SpanDetails model = new SpanDetails(this.parentTraceId
                , this.spanId
                , this.parentSpanId
                , this.startTime
                , this.endTime
                , this.kind
                , this.spanName
                , this.attributes
                , this.events
                , this.status);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(SpanDetails model) {
                this.parentTraceId(model.getParentTraceId());
    this.spanId(model.getSpanId());
    this.parentSpanId(model.getParentSpanId());
    this.startTime(model.getStartTime());
    this.endTime(model.getEndTime());
    this.kind(model.getKind());
    this.spanName(model.getSpanName());
    this.attributes(model.getAttributes());
    this.events(model.getEvents());
    this.status(model.getStatus());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * trace id
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parentTraceId")
    private final String parentTraceId;

        /**
     * trace id
     * @return the value
     **/
    
    public String getParentTraceId() {
        return parentTraceId;
    }


        /**
     * span id
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("spanId")
    private final String spanId;

        /**
     * span id
     * @return the value
     **/
    
    public String getSpanId() {
        return spanId;
    }


        /**
     * parent span id
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parentSpanId")
    private final String parentSpanId;

        /**
     * parent span id
     * @return the value
     **/
    
    public String getParentSpanId() {
        return parentSpanId;
    }


        /**
     * start time
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("startTime")
    private final Long startTime;

        /**
     * start time
     * @return the value
     **/
    
    public Long getStartTime() {
        return startTime;
    }


        /**
     * end time
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("endTime")
    private final Long endTime;

        /**
     * end time
     * @return the value
     **/
    
    public Long getEndTime() {
        return endTime;
    }


        /**
     * type of span
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("kind")
    private final String kind;

        /**
     * type of span
     * @return the value
     **/
    
    public String getKind() {
        return kind;
    }


        /**
     * span name
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("spanName")
    private final String spanName;

        /**
     * span name
     * @return the value
     **/
    
    public String getSpanName() {
        return spanName;
    }


        /**
     * span attributes
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("attributes")
    private final java.util.Map<String, Object> attributes;

        /**
     * span attributes
     * @return the value
     **/
    
    public java.util.Map<String, Object> getAttributes() {
        return attributes;
    }


        /**
     * collections of trace objects
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("events")
    private final java.util.List<TraceEvent> events;

        /**
     * collections of trace objects
     * @return the value
     **/
    
    public java.util.List<TraceEvent> getEvents() {
        return events;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("status")
    private final SpanStatus status;

    
    public SpanStatus getStatus() {
        return status;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("SpanDetails(");
        sb.append("parentTraceId=").append(String.valueOf(this.parentTraceId));
        sb.append(", spanId=").append(String.valueOf(this.spanId));
        sb.append(", parentSpanId=").append(String.valueOf(this.parentSpanId));
        sb.append(", startTime=").append(String.valueOf(this.startTime));
        sb.append(", endTime=").append(String.valueOf(this.endTime));
        sb.append(", kind=").append(String.valueOf(this.kind));
        sb.append(", spanName=").append(String.valueOf(this.spanName));
        sb.append(", attributes=").append(String.valueOf(this.attributes));
        sb.append(", events=").append(String.valueOf(this.events));
        sb.append(", status=").append(String.valueOf(this.status));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof SpanDetails)) {
            return false;
        }

        SpanDetails other = (SpanDetails) o;
        return java.util.Objects.equals(this.parentTraceId, other.parentTraceId) &&
            java.util.Objects.equals(this.spanId, other.spanId) &&
            java.util.Objects.equals(this.parentSpanId, other.parentSpanId) &&
            java.util.Objects.equals(this.startTime, other.startTime) &&
            java.util.Objects.equals(this.endTime, other.endTime) &&
            java.util.Objects.equals(this.kind, other.kind) &&
            java.util.Objects.equals(this.spanName, other.spanName) &&
            java.util.Objects.equals(this.attributes, other.attributes) &&
            java.util.Objects.equals(this.events, other.events) &&
            java.util.Objects.equals(this.status, other.status);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.parentTraceId == null ? 43 : this.parentTraceId.hashCode());
        result = (result * PRIME) + (this.spanId == null ? 43 : this.spanId.hashCode());
        result = (result * PRIME) + (this.parentSpanId == null ? 43 : this.parentSpanId.hashCode());
        result = (result * PRIME) + (this.startTime == null ? 43 : this.startTime.hashCode());
        result = (result * PRIME) + (this.endTime == null ? 43 : this.endTime.hashCode());
        result = (result * PRIME) + (this.kind == null ? 43 : this.kind.hashCode());
        result = (result * PRIME) + (this.spanName == null ? 43 : this.spanName.hashCode());
        result = (result * PRIME) + (this.attributes == null ? 43 : this.attributes.hashCode());
        result = (result * PRIME) + (this.events == null ? 43 : this.events.hashCode());
        result = (result * PRIME) + (this.status == null ? 43 : this.status.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/span_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class SpanDetails(object):
    """
    Span details
    """

    def __init__(self, **kwargs):
        """
        Initializes a new SpanDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param parent_trace_id:
            The value to assign to the parent_trace_id property of this SpanDetails.
        :type parent_trace_id: str

        :param span_id:
            The value to assign to the span_id property of this SpanDetails.
        :type span_id: str

        :param parent_span_id:
            The value to assign to the parent_span_id property of this SpanDetails.
        :type parent_span_id: str

        :param start_time:
            The value to assign to the start_time property of this SpanDetails.
        :type start_time: int

        :param end_time:
            The value to assign to the end_time property of this SpanDetails.
        :type end_time: int

        :param kind:
            The value to assign to the kind property of this SpanDetails.
        :type kind: str

        :param span_name:
            The value to assign to the span_name property of this SpanDetails.
        :type span_name: str

        :param attributes:
            The value to assign to the attributes property of this SpanDetails.
        :type attributes: dict(str, object)

        :param events:
            The value to assign to the events property of this SpanDetails.
        :type events: list[oci.aidataplatform_dp.models.TraceEvent]

        :param status:
            The value to assign to the status property of this SpanDetails.
        :type status: oci.aidataplatform_dp.models.SpanStatus

        """
        self.swagger_types = {
            'parent_trace_id': 'str',
            'span_id': 'str',
            'parent_span_id': 'str',
            'start_time': 'int',
            'end_time': 'int',
            'kind': 'str',
            'span_name': 'str',
            'attributes': 'dict(str, object)',
            'events': 'list[TraceEvent]',
            'status': 'SpanStatus'
        }

        self.attribute_map = {
            'parent_trace_id': 'parentTraceId',
            'span_id': 'spanId',
            'parent_span_id': 'parentSpanId',
            'start_time': 'startTime',
            'end_time': 'endTime',
            'kind': 'kind',
            'span_name': 'spanName',
            'attributes': 'attributes',
            'events': 'events',
            'status': 'status'
        }

        self._parent_trace_id = None
        self._span_id = None
        self._parent_span_id = None
        self._start_time = None
        self._end_time = None
        self._kind = None
        self._span_name = None
        self._attributes = None
        self._events = None
        self._status = None

    @property
    def parent_trace_id(self):
        """
        **[Required]** Gets the parent_trace_id of this SpanDetails.
        trace id


        :return: The parent_trace_id of this SpanDetails.
        :rtype: str
        """
        return self._parent_trace_id

    @parent_trace_id.setter
    def parent_trace_id(self, parent_trace_id):
        """
        Sets the parent_trace_id of this SpanDetails.
        trace id


        :param parent_trace_id: The parent_trace_id of this SpanDetails.
        :type: str
        """
        self._parent_trace_id = parent_trace_id

    @property
    def span_id(self):
        """
        **[Required]** Gets the span_id of this SpanDetails.
        span id


        :return: The span_id of this SpanDetails.
        :rtype: str
        """
        return self._span_id

    @span_id.setter
    def span_id(self, span_id):
        """
        Sets the span_id of this SpanDetails.
        span id


        :param span_id: The span_id of this SpanDetails.
        :type: str
        """
        self._span_id = span_id

    @property
    def parent_span_id(self):
        """
        Gets the parent_span_id of this SpanDetails.
        parent span id


        :return: The parent_span_id of this SpanDetails.
        :rtype: str
        """
        return self._parent_span_id

    @parent_span_id.setter
    def parent_span_id(self, parent_span_id):
        """
        Sets the parent_span_id of this SpanDetails.
        parent span id


        :param parent_span_id: The parent_span_id of this SpanDetails.
        :type: str
        """
        self._parent_span_id = parent_span_id

    @property
    def start_time(self):
        """
        **[Required]** Gets the start_time of this SpanDetails.
        start time


        :return: The start_time of this SpanDetails.
        :rtype: int
        """
        return self._start_time

    @start_time.setter
    def start_time(self, start_time):
        """
        Sets the start_time of this SpanDetails.
        start time


        :param start_time: The start_time of this SpanDetails.
        :type: int
        """
        self._start_time = start_time

    @property
    def end_time(self):
        """
        **[Required]** Gets the end_time of this SpanDetails.
        end time


        :return: The end_time of this SpanDetails.
        :rtype: int
        """
        return self._end_time

    @end_time.setter
    def end_time(self, end_time):
        """
        Sets the end_time of this SpanDetails.
        end time


        :param end_time: The end_time of this SpanDetails.
        :type: int
        """
        self._end_time = end_time

    @property
    def kind(self):
        """
        **[Required]** Gets the kind of this SpanDetails.
        type of span


        :return: The kind of this SpanDetails.
        :rtype: str
        """
        return self._kind

    @kind.setter
    def kind(self, kind):
        """
        Sets the kind of this SpanDetails.
        type of span


        :param kind: The kind of this SpanDetails.
        :type: str
        """
        self._kind = kind

    @property
    def span_name(self):
        """
        **[Required]** Gets the span_name of this SpanDetails.
        span name


        :return: The span_name of this SpanDetails.
        :rtype: str
        """
        return self._span_name

    @span_name.setter
    def span_name(self, span_name):
        """
        Sets the span_name of this SpanDetails.
        span name


        :param span_name: The span_name of this SpanDetails.
        :type: str
        """
        self._span_name = span_name

    @property
    def attributes(self):
        """
        **[Required]** Gets the attributes of this SpanDetails.
        span attributes


        :return: The attributes of this SpanDetails.
        :rtype: dict(str, object)
        """
        return self._attributes

    @attributes.setter
    def attributes(self, attributes):
        """
        Sets the attributes of this SpanDetails.
        span attributes


        :param attributes: The attributes of this SpanDetails.
        :type: dict(str, object)
        """
        self._attributes = attributes

    @property
    def events(self):
        """
        **[Required]** Gets the events of this SpanDetails.
        collections of trace objects


        :return: The events of this SpanDetails.
        :rtype: list[oci.aidataplatform_dp.models.TraceEvent]
        """
        return self._events

    @events.setter
    def events(self, events):
        """
        Sets the events of this SpanDetails.
        collections of trace objects


        :param events: The events of this SpanDetails.
        :type: list[oci.aidataplatform_dp.models.TraceEvent]
        """
        self._events = events

    @property
    def status(self):
        """
        **[Required]** Gets the status of this SpanDetails.

        :return: The status of this SpanDetails.
        :rtype: oci.aidataplatform_dp.models.SpanStatus
        """
        return self._status

    @status.setter
    def status(self, status):
        """
        Sets the status of this SpanDetails.

        :param status: The status of this SpanDetails.
        :type: oci.aidataplatform_dp.models.SpanStatus
        """
        self._status = status

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
